### DISCUSSION TITLE
Search Node P2V - Keep ES data

### DISCUSSION THREAD

**Discussion by 3isenHeiM** [Security-Onion-Solutions/securityonion#9820] [Unsupported Versions]
I need to perform a P2V transition of my 2 search nodes.
What's the best way to tackle this, without losing data (if possible).

I planned to register 2 new virtual search nodes, and then de-comission the 2 physical search nodes.

How to make sure that data in these search nodes is moved to the 2 new ones ?

Thanks.

**Comment by cm-ops** [comment]
If they are part of the same Elastic cluster, you could use https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-reroute.html. If cross cluster, you could snapshot and restore, or if time is not critical in shuttering the servers, install virtual search nodes and disable Logstash of physical servers to let the data age out, then shut them down.

**Reply by 3isenHeiM** [reply]
Well I shifted to turning them to true cluster first (see #9822), and then perform the P2V migration. We must secure the data replication first.

Thanks for the pointers though!
